### PR TITLE
ASN.1 Integer sequence

### PR DESCRIPTION
**Pull request by embedded-specialties**
# Description
Added integer type to the ASN.1 Sequencing and corrected comment typo

# Testing
Updated unit test
Tested 4 sets of two-integer sequences designed to test edge cases in the encodings

# Checklist

 - [X] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by embedded-specialties** [comment]
Hi WolfSSL,

I very that I submitted this pull request.

Thanks,
Joe

**Comment by anhu** [comment]
Fixes ZD 15939.

**Comment by embhorn** [comment]
Hello Joe,

We'll need to get a signed contributor agreement approved to accept changes from third parties. Could you please send a request for the contributor agreement form to our support email?
support@wolfssl.com

Thanks,
@embhorn

**Comment by embedded-specialties** [comment]
Done.

**Comment by embhorn** [comment]
@embedded-specialties is approved as a wolfSSL contributor.

**Comment by SparkiDev** [comment]
retest this please

File: src/ssl_asn1.c [hunk 1]
```diff
@@ -225,6 +231,15 @@ static int wolfssl_i2d_asn1_item(void** item, int type, byte* buf)
             len = wolfSSL_i2d_ASN1_BIT_STRING(
                 *(const WOLFSSL_ASN1_BIT_STRING**)item, buf);
             break;
+        case WOLFSSL_ASN1_INTEGER_ASN1:
+            byte *tmp_buf = buf;
```
Comment (SparkiDev) [inline src/ssl_asn1.c:235]: Older compilers don't allow a declaration unless in a block.
Put braces around this case.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/ssl_asn1.c (hunk 1 above)
Comment (embedded-specialties) [reply src/ssl_asn1.c:236]: Done, thanks!

**Comment by SparkiDev** [comment]
retest this please

File: tests/api.c [hunk 2]
```diff
@@ -33606,12 +33607,16 @@ static int test_wolfSSL_IMPLEMENT_ASN1_FUNCTIONS(void)
...
 
+    /* Test integer */
+    AssertNotNull(test_asn1 = TEST_ASN1_new());
+    der = NULL;
+    AssertIntEQ(i2d_TEST_ASN1(test_asn1, &der), 4);
+    XFREE(der, NULL, DYNAMIC_TYPE_ASN1);
+    TEST_ASN1_free(test_asn1);
+
     /* Test error cases. */
-    AssertNull(TEST_ASN1_new());
```
Comment (SparkiDev) [inline tests/api.c:33610]: Restore the failure case.
Change it to be an OCTET_STRING now that you have added support for INTEGER.

File: tests/api.c [hunk 3]
```diff
@@ -33606,12 +33607,16 @@ static int test_wolfSSL_IMPLEMENT_ASN1_FUNCTIONS(void)
     XFREE(der, NULL, DYNAMIC_TYPE_ASN1);
     DPP_BOOTSTRAPPING_KEY_free(bootstrap);
 
+    /* Test integer */
+    AssertNotNull(test_asn1 = TEST_ASN1_new());
+    der = NULL;
+    AssertIntEQ(i2d_TEST_ASN1(test_asn1, &der), 4);
```
Comment (SparkiDev) [inline tests/api.c:33613]: This test case is failing with: ./configure CC="clang -fsanitize=address,undefined -g" --enable-all --enable-debug

There is not an integer value put into test_asn1 which results in an error when wolfSSL_i2d_ASN1_INTEGER is called which messes up the length.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: tests/api.c (hunk 3 above)
Comment (embedded-specialties) [reply tests/api.c:33613]: Done.

**Comment by SparkiDev** [comment]
Hi @embedded-specialties

Can you please explain how you ended up doing this work?
Do you have a requirement in your application that requires that WOLFSSL_ASN1_INTEGER_ASN1 be supported?

Thanks,
Sean

**Comment by embedded-specialties** [comment]
Hi Sean,

Yes.  The application uses the sequence feature with ASN.1 Integers.

Thanks,
Joe

________________________________

**Review by anhu — APPROVED** [review]
LGTM

**Comment by anhu** [comment]
By the way, I had a hand in helping @embedded-specialties develop this PR so I don't think my review should count. 

Anthony.

**Review by SparkiDev — APPROVED** [review]

**Comment by SparkiDev** [comment]
retest this please
